common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -560,7 +564,9 @@ def should_force_account_creation():
         elif auth_entry == AUTH_ENTRY_REGISTER:
             # User has authenticated with the third party provider and now wants to finish
             # creating their edX account.
-            return dispatch_to_register()
+            register_origin_url = strategy.request.session.get('next')
+            return dispatch_to_custom_redirect(register_origin_url) \
```

Comment:
Can we redirect from here instead of function?

Reply:
@emraan004 goes with the flow of how it is previously written. If you still think we should redirect directly from here, I can make that change

Reply:
It felt unnecessary, but if this is how it was implemented, then it is fine for me. I'm resolving the conversation.